penty/pentypes/builtins.py

```diff
@@ -826,43 +830,236 @@ def dict_value_iterator_instanciate(ty):
 ##
 #
 
-def list_append(self, value_ty):
-    self.__args__[0].add(_astype(value_ty))
+def list_add(self, value):
+    if not issubclass(value, list):
```

Comment:
this check is no longer needed, right?

Reply:
This is `list.__add__`. The `MethodType` wrapper makes sure that `self` (a.k.a lhs) is a list, but we also need to make sure that value (a.k.a rhs) is also a list. `list() + 3` is invalid.